ipatests/test_integration/test_cert.py

```diff
@@ -127,6 +128,84 @@ def test_ipa_getcert_san_aci(self):
         ipaddrs = ext.value.get_values_for_type(x509.IPAddress)
         assert ipaddrs == [ipaddress.ip_address(self.clients[0].ip)]
 
+    def test_getcert_list_profile(self):
+        """
+        Test that getcert list command displays the profile
+        for the requests
+        """
+        for requests in os.listdir(paths.CERTMONGER_REQUESTS_DIR):
+            result = self.master.run_command(
+                ["getcert", "list", "-i", requests], raiseonerr=False
+            )
+            assert "profile: caSubsystemCert" in result.stdout_text
+            assert "profile: caSignedLogCert" in result.stdout_text
+            assert "profile: caOCSPCert" in result.stdout_text
+            assert "profile: caSubsystemCert" in result.stdout_text
+            assert "profile: caCACert" in result.stdout_text
+            assert "profile: caServerCert" in result.stdout_text
+            assert "profile: caIPAserviceCert" in result.stdout_text
+            assert "profile: caIPAserviceCert" in result.stdout_text
+            assert "profile: KDCs_PKINIT_Certs" in result.stdout_text
+
+    def test_getcert_list_profile_using_subca(self):
+        """
+        Test that getcert list command displays the profile
+        for the request with a SUBCA configured on IPA server
+        """
+        sub_name = "CN=SUBCA"
+        tasks.kinit_admin(self.master)
+        self.master.run_command(
+            ["ipa", "ca-add", "mysubca", "--subject=" + sub_name]
+        )
+        self.master.run_command(
+            [
+                "ipa",
+                "caacl-add-ca",
+                "hosts_services_caIPAserviceCert",
+                "--cas=mysubca",
+            ]
+        )
+        cmd_arg = [
+            "getcert",
+            "request",
+            "-c",
+            "ipa",
+            "-I",
+            "test-request",
+            "-k",
+            "/etc/pki/tls/private/test.key",
+            "-f",
+            "/etc/pki/tls/certs/test.pem",
+            "-D",
+            self.master.hostname,
+            "-K",
+            "host/%s" % self.master.hostname,
+            "-N",
+            "CN="+self.master.hostname,
+            "-U",
+            "id-kp-clientAuth",
+            "-X",
+            "mysubca",
+            "-T",
+            "caIPAserviceCert",
+        ]
+        result = self.master.run_command(cmd_arg)
+        assert (
+            'New signing request "test-request" added.\n' in result.stdout_text
+        )
+        status = tasks.wait_for_request(self.master, test-request, 50)
+        if status == "MONITORING":
+            result = self.master.run_command(
+                ["getcert", "list", "-i", "test-request"]
+            )
+            assert "profile: caIPAserviceCert" in result.stdout_text
```

Comment:
if status is not MONITORING then the test will pass. You'll need to do an assert at the end of the test to raise an issue if not.

If this test fails then it is going to leave the request and the CA behind which could cause other tests to fail. I suspect this is a good case for a fixture.

Reply:
You still need an else: <force failure via some sort of assert>

Reply:
Added the else statement as well.